authomatic/providers/oauth2.py

```diff
@@ -1200,6 +1200,38 @@ def _x_credentials_parser(cls, credentials, data):
             credentials.token_type = cls.BEARER
         return credentials
 
+    def access(self, url, **kwargs):
+        # https://developer.github.com/v3/#user-agent-required
+        # Github requries that all API requests MUST include a valid ``User-Agent`` header.
+        headers = kwargs["headers"] = kwargs.get("headers", {})
+        if not headers.get("User-Agent"):
+            headers["User-Agent"] = self.settings.config[self.name]["consumer_key"]
```

Comment:
Why do you use the consumer key as user agent? In the referenced documentation this is not mentioned, just to be some fixed string should be enough, like "Authomatic"?

Reply:
Because by using the `consumer_key` Github can identify the SP.
> We request that you use your GitHub username, or the name of your application, for the `User-Agent` header value. This allows us to contact you if there are problems.

https://developer.github.com/v3/#user-agent-required